app/config/drupal-clean/download.sh

```diff
@@ -18,7 +18,8 @@ pushd "$WEB_ROOT/web"
     git clone "${CACHE_DIR}/civicrm/civicrm-drupal.git"                      -b "7.x-$CIVI_VERSION"  civicrm/drupal
     git clone "${CACHE_DIR}/civicrm/civicrm-packages.git"                    -b "$CIVI_VERSION"      civicrm/packages
     git clone "${CACHE_DIR}/eileenmcnaughton/civicrm_developer.git"          -b master               civicrm_developer
-    git clone "${CACHE_DIR}/civicrm/api4.git"                                -b "master"             civicrm/ext/api4
+    EXTCIVIVER=$( php -r '$x=simplexml_load_file("civicrm/xml/version.xml"); echo $x->version_no;' )
+    cv dl -b "@https://civicrm.org/extdir/ver=$EXTCIVIVER|cms=Drupal|status=|ready=/org.civicrm.api4.xml" --to="$WEB_ROOT/web/sites/all/modules/civicrm/ext/api4" --dev
```

Comment:
Right, this makes sense for like 90% of cases. The exceptions are:

* PR testing of `org.civicrm.api4.git`
* Ext matrix testing focused on `org.civicrm.api4`

Here's an interesting observation - consider the steps when running those:

```
DEBUG=1 ~/src/civici/bin/civici ext:build -v -v -v --pr-url=https://github.com/civicrm/org.civicrm.api4/pull/191 --dry-run --build=pr123 --build-root ~/bknix/build/

Download main codebase (build=pr123, type=drupal-clean, civi-ver=master)
$ cd '/Users/totten/bknix/build/dmaster/web/sites/all/modules/civicrm/ext/api4'
$ civibuild download 'pr123' --type 'drupal-clean' --civi-ver 'master'

Download extension PR (https://github.com/civicrm/org.civicrm.api4/pull/191)
$ cd '/Users/totten/bknix/build/pr123'
$ git clonepr --merged 'https://github.com/civicrm/org.civicrm.api4/pull/191' 'web/sites/default/files/civicrm/ext/target' --depth 1

Download extension dependencies
$ cd '/Users/totten/bknix/build/pr123'
$ civici ext:dl-dep --info='web/sites/default/files/civicrm/ext/target'/info.xml --feed='https://civicrm.org/extdir/ver=5.40.0|uf=Bare|status=|ready=/single' --to='/Users/totten/bknix/build/pr123/web/sites/default/files/civicrm/ext'

Install main database
$ cd '/Users/totten/bknix/build/pr123'
$ civibuild install 'pr123'
```

Since `drupal-clean/download.sh` now pulls in `api4`, the test build is actually getting two copies of `api4` (once via `drupal-clean/download.sh` and once via `git clonepr`), so test runs of `api4` are currently relying on an undefined behavior...